mkosi/config.py

```diff
@@ -1765,6 +1773,36 @@ def backward_compat_stubs(self, namespace: argparse.Namespace) -> None:
             delattr(namespace, "cache")
             print("Warning: --cache is no longer supported")
 
+    def resolve_deps(self, args: MkosiArgs, presets: Sequence[MkosiConfig]) -> list[MkosiConfig]:
+        graph = {p.preset: p.depends for p in presets}
+
+        if args.presets:
+            for p in args.presets:
+                if p not in graph:
+                    die(f"No preset found with name {p}")
+
+            deps = set()
+            queue = [*args.presets]
+
+            while queue:
+                preset = queue.pop()
+                if preset in deps:
+                    continue
+
+                deps.add(preset)
+                queue += [*graph[preset]]
+
+            presets = [p for p in presets if p.preset in deps]
+
+        graph = {p.preset: p.depends for p in presets}
```

Comment:
So, this rebuilds `graph` with only the members of `presets` in `deps`, right? This should be equivalent to removing all nodes from `graph` that are not in `deps`, i.e.
```suggestion
        graph = {node: graph[node] for node in graph.keys() & deps}
```

This doesn't preserve the order of the keys in `graph`, but this doesn't matter, because the order will be changed by the topological sort right atferwards anyway.

Reply:
Yes but we also need these removed from presets for the sort we do later

Reply:
This isn't sufficient, we need to filter the preset list as well so we can sort it properly later.

Reply:
Ah, true. Maybe it could be more succinct by using the presets themselves as keys in the graph instead of their names, but I think it's fine as is.